Header header
bool is_stable